packages/web-components/src/button/button.styles.ts

```diff
@@ -0,0 +1,321 @@
+  }
+
+  :host([appearance='outline']:hover:active) .control {
+    background-color: ${colorTransparentBackgroundPressed};
+  }
+
+  :host(is:([disabled][appearance='outline'], [disabled-focusabale][appearance="outline"])) .control,
+  :host(is:([disabled][appearance='outline'], [disabled-focusabale][appearance="outline"]):hover) .control,
+  :host(is:([disabled][appearance='outline'], [disabled-focusabale][appearance="outline"]):hover:active) .control {
+    background-color: ${colorTransparentBackground};
+  }
+
+  :host([appearance='subtle']) .control {
+    background-color: ${colorSubtleBackground};
+    color: ${colorNeutralForeground2};
+    border-color: transparent;
+  }
+
+  :host([appearance='subtle']:hover) .control {
+    background-color: ${colorSubtleBackgroundHover};
+    color: ${colorNeutralForeground2Hover};
+    border-color: transparent;
+  }
+
+  :host([appearance='subtle']:hover:active) .control {
+    background-color: ${colorSubtleBackgroundPressed};
+    color: ${colorNeutralForeground2Pressed};
+    border-color: transparent;
+  }
+
+  :host(is:([disabled][appearance='subtle'], [disabled-focusabale][appearance="subtle"])) .control,
+  :host(is:([disabled][appearance='subtle'], [disabled-focusabale][appearance="subtle"]):hover) .control,
+  :host(is:([disabled][appearance='subtle'], [disabled-focusabale][appearance="subtle"]):hover:active) .control {
+    background-color: ${colorTransparentBackground};
+    border-color: transparent;
+  }
+
+  :host([appearance='subtle']:hover) ::slotted(svg) {
+    fill: ${colorNeutralForeground2BrandHover};
+  }
+
+  :host([appearance='subtle']:hover:active) ::slotted(svg) {
+    fill: ${colorNeutralForeground2BrandPressed};
+  }
+
+  :host([appearance='transparent']) .control {
+    background-color: ${colorTransparentBackground};
+    color: ${colorNeutralForeground2};
+  }
+
+  :host([appearance='transparent']:hover) .control {
+    background-color: ${colorTransparentBackgroundHover};
+    color: ${colorNeutralForeground2BrandHover};
+  }
+
+  :host([appearance='transparent']:hover:active) .control {
+    background-color: ${colorTransparentBackgroundPressed};
+    color: ${colorNeutralForeground2BrandPressed};
+  }
+
+  :host([appearance='transparent']) .control,
+  :host([appearance='transparent']:hover) .control,
+  :host([appearance='transparent']:hover:active) .control {
+    border-color: transparent;
+  }
+
+  :host(is:([disabled][appearance='transparent'], [disabled-focusabale][appearance="transparent"])) .control,
+  :host(is:([disabled][appearance='transparent'], [disabled-focusabale][appearance="transparent"]):hover) .control,
+  :host(is:([disabled][appearance='transparent'], [disabled-focusabale][appearance="transparent"]):hover:active) .control {
+    border-color: transparent;
+    background-color: ${colorTransparentBackground};
+  }
+
+  :host(:is([disabled], [disabled-focusable], [appearance][disabled], [appearance][disabled-focusable])) .control,
```

Comment:
Could you please explain why the appearance part of the selector (`[appearance][disabled], [appearance][disabled-focusable]`) is necessary?

Reply:
specificity - I can try and pull it off, but we have a number of styles where we're changing the various button states themselves, many which use attribute selectors. This ensures that regardless of the actual attribute value, ALL appearances should have these styles applied in both disabled and disabled-focusable scenarios. It's possible that :is() provides enough specificity...but I'm not sure. I can test when I get the repo built again.